### PR TITLE
Anal

### PR DESCRIPTION
**Pull request by alvarofe**
Initial attempt to fix fcn->_size and related. 

I removed some checks in r_anal_fcn when overlapping, is that ok?

Thoughts about removing r_anal_fcn_set_size and the command af+? I think the way to define function by hand should be through basic blocks. Its used in FITFCNSZ for example as well.

I checked against dll from https://github.com/radare/radare2/issues/5718 and it seems there is an improvement with the frames. This is not meant to be merged since there will be many thing to test and fix (RAnalFunction still has _size parameter) ... just food for thought.

### CODE REVIEW

**Comment by Maijin** [comment]
Regressions

**Comment by alvarofe** [comment]
Obviously haha this is gonna break a lot of stuff

**Comment by alvarofe** [comment]
The aim of this is to define new API I didn't look to the test at all

**Comment by Maijin** [comment]
https://github.com/radare/radare2/issues/4868

```
Remove concept of realsize and size, there is only one size, and it's the sum of the sizes of the basic blocks the function is composed of
```

**Comment by radare** [comment]
Thinking a bit more on this i think it makes sense to keep size but those have diferent meanings:

* symbol size = exposed by rbin or user defined
* sum all bb sizes = code coverage
* boundaries(range) = max - min

**Comment by radare** [comment]
There are a bunch of sizes in fact and i think we should honor them all

**Comment by alvarofe** [comment]
Fix anal as it is in r2 may be a harsh cause the craziness of the code. I think that aab and a2f come in handy here. `aab` is quite new and it can be developed having all this concepts in mind (plus @radare says it works well where `af` fails in some situations) and `a2f` is using `sdb` already, the code is quite clear and extend it is way easier than the loop in fcn.c at least imho. 

What do you think? it will pay off to rewrite anal completely or it would be better to start with `a2f`?

**Comment by Maijin** [comment]
Before getting nuts on anal, I'd strongly go for finishing SIOL.

**Comment by radare** [comment]
This was a playground for the fcn size refactoring needed to make travis happy

File: libr/anal/fcn.c [hunk 1]
```diff
@@ -256,14 +256,6 @@ static RAnalBlock* appendBasicBlock (RAnal *anal, RAnalFunction *fcn, ut64 addr)
 	return bb;
 }
 
-#define FITFCNSZ() {\
```
Comment (radare) [inline libr/anal/fcn.c]: try doing this PR without removing _size

File: libr/core/cconfig.c [hunk 2]
```diff
@@ -2054,6 +2065,7 @@ R_API int r_core_config_init(RCore *core) {
 
 	/* hexdump */
 	SETCB("hex.pairs", "true", &cb_hexpairs, "Show bytes paired in 'px' hexdump");
+	SETCB("hex.compact", "false", &cb_hexcompact, "Show smallest 16 byte col hexdump (60 columns)");
```
Comment (radare) [inline libr/core/cconfig.c]: wat

**Comment by radare** [comment]
please rebase, there are conflicts. also, maybe we shuold fix this without removing the FITFCNSZ or removing _size. it is at the end needed for the current analysis. and greenifying was only about having symsize imho

there's also things like this:

```
-|       |   0x000101a4      ef00002d       jal ra, sym.puts
+|       |   0x000101a4      ef00002d       jal ra, sym.puts           ; sym.printf+0x100
```

**Comment by alvarofe** [comment]
Having symsize won't fix anything or I don't see it. _size is related with bbs and adding symsize we should add new logic that would make it useful for something. Like for example in FITFCNSZ iterate over flags and grab the symbol size at that address and assign it to the function. Morever, the symbol size changes cause the flag is updated when defining a function.

```
[0x100001174]> f~sym.func.1000043ff
0x1000043ff 0 sym.func.1000043ff
[0x100001174]> s sym.func.1000043ff
[0x1000043ff]> af
[0x1000043ff]> f~sym.func.1000043ff
0x1000043ff 46 sym.func.1000043ff
```

**Comment by alvarofe** [comment]
So i would drop this and I would work on spare times on a2f to improve it.

**Comment by radare** [comment]
imho this PR without the fitfcnsz will make travis green again because functions will display the symsize

**Comment by radare** [comment]
i would like functions to store the symsize if found before analyzing. even if this info is not used it should be displayed in afi. also we have a problem with ascii art and function boundaries.. but thats another issue

**Comment by alvarofe** [comment]
I wouldn't add that cause at this stage it will add another lookup for symbols and might introduce slowness and it does not add a value imho. I will add this though once flags has been moved into ht, making the lookup fastest.

**Comment by radare** [comment]
ping

**Pull request closed without merging**